cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -179,30 +180,98 @@ def __init__(self, scope: core.Construct, id: str, head_node: HeadNode, **kwargs
-        super().__init__(scope, id)
-        self.fsx = fsx
-        # TODO add all the other required info other than fsx object and generate template
-        # TODO Verify if there are building blocks ready to be used
-
-
 class ClusterStack(core.Stack):
     """Create the Stack and delegate to specific Construct for the creation of all the resources for the Cluster."""
 
     def __init__(self, scope: core.Construct, construct_id: str, cluster: SlurmCluster, **kwargs) -> None:
         super().__init__(scope, construct_id, **kwargs)
         self._cluster = cluster
-
+        self._storage_resource_ids = {SharedStorage.Type.EBS: [], SharedStorage.Type.FSX: []}
         HeadNodeConstruct(self, "HeadNode", cluster.head_node)
+
         if cluster.shared_storage:
             for storage in cluster.shared_storage:
-                if storage.fsx:
-                    FsxConstruct(self, "Fsx", storage.fsx)
-                # if storage.efs:
-                #    EfsConstruct(storage.efs)  # TODO Verify if there are building blocks ready to be used
-                # if storage.ebs:
-                #    EbsConstruct(ebs)  # Verify if there are building blocks ready to be used
+                self._add_shared_storage(storage)
+
+        self._add_shared_storage_outputs()
+
+    def _add_shared_storage(self, storage: SharedStorage):
+        """Add specific Cfn Resources to map the shared storage and store the output filesystem id."""
+        storage_id = None
+        cfn_resource_id = "{0}{1}".format(
+            storage.shared_storage_type.name, len(self._storage_resource_ids[storage.shared_storage_type])
+        )
+        if storage.shared_storage_type == SharedStorage.Type.FSX:
+            storage_id = self._add_fsx_storage(
+                cfn_resource_id,
+                storage,
+                subnet_ids=self._cluster.compute_subnet_ids,
+                security_group_ids=self._cluster.compute_security_groups,
+            )
+        elif storage.shared_storage_type == SharedStorage.Type.EBS:
+            storage_id = self._add_ebs_volume(cfn_resource_id, storage)
+
+        self._storage_resource_ids[storage.shared_storage_type].append(storage_id)
+
+    def _add_shared_storage_outputs(self):
+        """Add the ids of the managed filesystem to the Stack Outputs."""
+        for storage_type, storage_ids in self._storage_resource_ids.items():
+            core.CfnOutput(
+                scope=self,
+                id="{0}Ids".format(storage_type.name),
+                description="{0} Filesystem IDs".format(storage_type.name),
+                value=",".join(storage_ids),
+            )
+
+    def _add_fsx_storage(self, id: str, shared_fsx: SharedFsx, subnet_ids: [str], security_group_ids: [str]):
+        """Add specific Cfn Resources to map the FSX storage."""
+        fsx_id = shared_fsx.file_system_id.value
+
+        if not fsx_id and shared_fsx.mount_dir:
```

Comment:
`shared_fsx.mount_dir` is Param, so should we check it value?

Reply:
Right! Fixed now